### PR TITLE
Datagrid: Fix untyped code

### PR DESCRIPTION
**Pull request by joshhunt**
Fixes typescript errors in Datagrid that managed to slip in recently.

I've done a very brief smoketest of the panel and it all seems to work, but I'm unsure how to trigger the text boxes (where there is an actual change in code)

### CODE REVIEW

File: public/app/plugins/panel/datagrid/state.ts
Comment (torkelo) [inline public/app/plugins/panel/datagrid/state.ts:136]: All of these types are unnecessary if we used redux toolkit

Comment (torkelo) [reply public/app/plugins/panel/datagrid/state.ts:136]: except the payload types :)

Comment (joshhunt) [reply public/app/plugins/panel/datagrid/state.ts:136]: Completely true - this probably would be a whole lot easier with redux toolkit, but I'll leave that up to the code owners to do. I just wanted to come in and fix these with some quick and trival changes :)

Comment (mdvictor) [reply public/app/plugins/panel/datagrid/state.ts:136]: Agreed, it's on my todolist to refactor this in a better way

**Review by mdvictor — APPROVED** [review]
Looks and works great. Thank you so much @joshhunt!

**Pull request merged**
